### PR TITLE
[mlir][Vector] Move insert/extractelement distribution patterns to insert/extract

### PR DESCRIPTION
**Pull request by Groverkss**
This is a NFC-ish change that moves vector.extractelement/vector.insertelement vector distribution patterns to vector.insert/vector.extract.

Before:

0-d/1-d vector.extract -> vector.extractelement -> distributed vector.extractelement
2-d+ vector.extract -> distributed vector.extract

After:

scalar input vector.extract -> distributed vector.extract
vector.extractelement -> distributed vector.extract
2d+ vector.extract -> distributed vector.extract

The same changes are done for insertelement/insert. The change allows us to remove reliance on vector.extractelement/vector.insertelement, which are soon to be depreciated: https://discourse.llvm.org/t/rfc-psa-remove-vector-extractelement-and-vector-insertelement-ops-in-favor-of-vector-extract-and-vector-insert-ops/71116/8

No extra tests are included because this patch doesn't introduce / remove any functionality. It only changes the chain of lowerings. This change can be completly NFC if we make the distributed operation vector.extractelement/vector.insertelement, but that is slightly weird, because you are going from extractelement -> extract -> extractelement.

### CODE REVIEW

**Review by dcaballe — APPROVED** [review]
Thanks for the cleanup! Please, give time to others to review.

File: mlir/lib/Dialect/Vector/Transforms/VectorDistribute.cpp [hunk 1]
```diff
@@ -1413,31 +1397,60 @@ struct WarpOpExtractElement : public OpRewritePattern<WarpExecuteOnLane0Op> {
...
+    if (!operand)
+      return failure();
+    auto extractOp = operand->get().getDefiningOp<vector::ExtractElementOp>();
+    rewriter.setInsertionPoint(extractOp);
+    if (auto pos = extractOp.getPosition()) {
+      rewriter.replaceOpWithNewOp<vector::ExtractOp>(
+          extractOp, extractOp.getVector(), pos);
+    } else {
+      rewriter.replaceOpWithNewOp<vector::ExtractOp>(
+          extractOp, extractOp.getVector(), ArrayRef<int64_t>{});
```
[the remark this answers is not in the document]
Comment (banach-space) [reply mlir/lib/Dialect/Vector/Transforms/VectorDistribute.cpp:1417]: +1

**Review by banach-space — APPROVED** [review]
Makes sense, much appreciated step on our journey towards:
* [https://discourse.llvm.org/t/rfc-psa-remove-vector-extractelement-and-vector-insertelement-ops-in-favor-of-vector-extract-and-vector-insert-ops](https://discourse.llvm.org/t/rfc-psa-remove-vector-extractelement-and-vector-insertelement-ops-in-favor-of-vector-extract-and-vector-insert-ops/71116/8)

Changes in test make it clear that this is NFC, LGTM.

File: mlir/lib/Dialect/Vector/Transforms/VectorDistribute.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (Groverkss) [reply mlir/lib/Dialect/Vector/Transforms/VectorDistribute.cpp]: done

**Pull request merged**